### PR TITLE
[complex32] conj and conj_physical

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference: https://github.com/pytorch/pytorch/issues/74537

Required for `complex32` support by `fft` module.

Tested with `test_dtypes` and `test_complex_half_reference_testing` in test_ops.py

### CODE REVIEW

File: aten/src/ATen/cpu/vec/vec_base.h [hunk 1]
```diff
@@ -133,7 +133,7 @@ struct Vectorized {
   static constexpr size_type size() {
     return VECTOR_WIDTH / sizeof(T);
   }
-  Vectorized() : values{0} {}
+  Vectorized() : values{static_cast<T>(0)} {}
```
Comment (kshitij12345) [inline aten/src/ATen/cpu/vec/vec_base.h:136]: We allow only explicit construction of `complex32` at this point.

File: aten/src/ATen/native/cuda/UnaryComplexKernels.cu [hunk 2]
```diff
@@ -81,13 +82,26 @@ __host__ __device__ static inline c10::complex<T> conj_wrapper(c10::complex<T> v
...
+  auto common_dtype = iter.common_dtype();
+  if (common_dtype == kComplexHalf) {
+    using scalar_t = c10::complex<at::Half>;
+    static const auto conj_string = jiterator_stringify(
+      template <typename T>
+      T conj_kernel(T z) {
+        return std::conj(z);
+      }
+    );
+    jitted_gpu_kernel<conj_name, scalar_t, scalar_t, 1>(iter, conj_string);
```
Comment (kshitij12345) [inline aten/src/ATen/native/cuda/UnaryComplexKernels.cu:96]: jiterate only on `complex32`

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -9422,12 +9422,17 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
                    ),
     UnaryUfuncInfo('conj',
                    ref=np.conj,
-                   dtypes=all_types_and_complex_and(torch.bool,
-                                                    torch.bfloat16, torch.half),
+                   dtypes=all_types_and_complex_and(torch.bool, torch.bfloat16,
+                                                    torch.half, torch.complex32),
```
Comment (anjali411) [inline torch/testing/_internal/common_methods_invocations.py:9426]: you probably wanna enable this test for `conj_physical` which actually performs the conj computation

**Review by anjali411 — APPROVED** [review]

**Comment by anjali411** [comment]
The test failure looks real
```
======================================================================
ERROR [0.076s]: test_reference_numerics_normal_conj_physical_cpu_complex32 (__main__.TestUnaryUfuncsCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/site-packages/torch/testing/_internal/common_device_type.py", line 376, in instantiated_test
    result = test(self, **param_kwargs)
  File "/opt/conda/lib/python3.7/site-packages/torch/testing/_internal/common_device_type.py", line 773, in test_wrapper
    return test(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py", line 1145, in wrapper
    fn(*args, **kwargs)
  File "test_unary_ufuncs.py", line 352, in test_reference_numerics_normal
    self._test_reference_numerics(dtype, op, tensors)
  File "test_unary_ufuncs.py", line 321, in _test_reference_numerics
    a = t.cpu().numpy()
TypeError: Got unsupported ScalarType ComplexHalf

----------------------------------------------------------------------
Ran 8303 tests in 317.490s
```

**Comment by kshitij12345** [comment]
@anjali411 Thanks! Have added an expectedFailure. The test fails as `.numpy()` fails as there is no corresponding dtype for `complex32`.

**Comment by kshitij12345** [comment]
There are a few failures in `sparse` test suite for `conj_physical` due to few operation used for test not supporting `ComplexHalf`. Will add expectedFailures.

**Comment by kshitij12345** [comment]
@anjali411 this PR is ready for merging

**Pull request closed without merging**